### PR TITLE
SDMMC frequency selection based on board type

### PR DESCRIPTION
**Pull request by dukess**
On Olimex ESP32 EVB I/O operations with SD card can cause error when LAN is used in same time.
Problem is disappearing if SD MMC frequency lower down from SDMMC_FREQ_HIGHSPEED to SDMMC_FREQ_DEFAULT.

No problem if WiFi used instead LAN.

For issue https://github.com/espressif/arduino-esp32/issues/5682

### CODE REVIEW

File: variants/esp32-evb/pins_arduino.h [hunk 1]
```diff
@@ -29,5 +29,6 @@ static const uint8_t MISO  = 15;
 static const uint8_t SCK   = 14;
 
 #define BOARD_HAS_1BIT_SDMMC
+#define ETH_CAN_INTERFERE_WITH_SDMMC
```
Comment (igrr) [inline variants/esp32-evb/pins_arduino.h:32]: Perhaps defining something like `BOARD_MAX_SDMMC_FREQ` would be a bit more generic? Then in sdmmc library you can include pins_arduino.h, check if BOARD_MAX_SDMMC_FREQ isn't defined, and define it to the default value. That would eliminate the ifdef around the begin function.

**Review by igrr — CHANGES_REQUESTED** [review]

File: variants/esp32-evb/pins_arduino.h (hunk 1 above)
Comment (dukess) [reply variants/esp32-evb/pins_arduino.h:32]: Thanks for idea. Will try it right now.

**Comment by dukess** [comment]
this pull request include this request too: https://github.com/espressif/arduino-esp32/pull/5631

File: variants/esp32-evb/pins_arduino.h (hunk 1 above)
Comment (dukess) [reply variants/esp32-evb/pins_arduino.h]: @igrr Sorry for disturb, i do not know how to correct declare about changes, so just calling you in this PR.

**Pull request merged**